### PR TITLE
LVA: Coding style cleanup

### PR DESCRIPTION
**Pull request by mary-georgiou-sonarsource**
Just taking the dust - no functional changes or other improvements 🧹 🧹 🧹 🧹 🧹 🧹 
I'll make meaningful functional cleanup changes, if needed, during actual work.

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/LiveVariableAnalysisBase.cs [hunk 1]
```diff
@@ -20,95 +20,94 @@
...
-    {
-        protected readonly ISymbol originalDeclaration;
-        protected readonly CancellationToken Cancel;
-        private readonly IDictionary<TBlock, HashSet<ISymbol>> blockLiveOut = new Dictionary<TBlock, HashSet<ISymbol>>();
-        private readonly IDictionary<TBlock, HashSet<ISymbol>> blockLiveIn = new Dictionary<TBlock, HashSet<ISymbol>>();
-        private readonly HashSet<ISymbol> captured = new();
+    protected readonly ISymbol originalDeclaration;
+    protected readonly CancellationToken Cancel;
+    private readonly IDictionary<TBlock, HashSet<ISymbol>> blockLiveOut = new Dictionary<TBlock, HashSet<ISymbol>>();
+    private readonly IDictionary<TBlock, HashSet<ISymbol>> blockLiveIn = new Dictionary<TBlock, HashSet<ISymbol>>();
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/LiveVariableAnalysisBase.cs:30]:
```suggestion
    private readonly Dictionary<TBlock, HashSet<ISymbol>> blockLiveOut = new();
    private readonly Dictionary<TBlock, HashSet<ISymbol>> blockLiveIn = new();
```

File: analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/LiveVariableAnalysisBase.cs [hunk 2]
```diff
@@ -20,95 +20,94 @@
...
-        /// <summary>
-        /// LiveOut variables are alive when exiting block. They are read in any of it's successors.
-        /// </summary>
-        public IReadOnlyCollection<ISymbol> LiveOut(TBlock block) =>
-            blockLiveOut[block].Except(captured).ToImmutableArray();
+    /// <summary>
+    /// LiveOut variables are alive when exiting block. They are read in any of it's successors.
+    /// </summary>
+    public IReadOnlyCollection<ISymbol> LiveOut(TBlock block) =>
+        blockLiveOut[block].Except(captured).ToImmutableArray();
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/LiveVariableAnalysisBase.cs:60]: `IReadOnlyCollection` and `ToImmutableArray` are overkills. All usages do some linq or hashset with it => return `IEnumerable<ISymbol>`, without alternation

File: analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/LiveVariableAnalysisBase.cs [hunk 3]
```diff
@@ -20,95 +20,94 @@
...
-            public ISet<ISymbol> Captured { get; } = new HashSet<ISymbol>();
-
-            protected State() { }
+            throw new InvalidOperationException("Out of exit block should be empty");
         }
     }
+
+    protected class State
+    {
+        public ISet<ISymbol> Assigned { get; } = new HashSet<ISymbol>();            // Kill: The set of variables that are assigned a value.
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/LiveVariableAnalysisBase.cs:106]: Make them all `HashSet` and use `new()` to initialize

File: analyzers/src/SonarAnalyzer.CSharp/LiveVariableAnalysis/SonarCSharpLiveVariableAnalysis.cs [hunk 4]
```diff
@@ -21,212 +21,211 @@
...
+                        break;
+
+                    case SyntaxKind.VariableDeclarator:
+                        ProcessVariableDeclarator((VariableDeclaratorSyntax)instruction);
+                        break;
+
+                    case SyntaxKind.AnonymousMethodExpression
+                    or SyntaxKind.ParenthesizedLambdaExpression
+                    or SyntaxKind.SimpleLambdaExpression
+                    or SyntaxKind.QueryExpression:
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/LiveVariableAnalysis/SonarCSharpLiveVariableAnalysis.cs:107]: The old `case` was better here due to alignment.

File: analyzers/tests/SonarAnalyzer.Test/AnalysisContext/SonarAnalysisContextBaseTest.ShouldAnalyzeTree.cs [hunk 5]
```diff
@@ -66,7 +66,7 @@ public void ShouldAnalyzeTree_Scanner_UnchangedFiles_ContainsTreeFile()
     [TestMethod]
     public void ShouldAnalyzeTree_Scanner_UnchangedFiles_ContainsOtherFile()
     {
-        var options = CreateOptions(new[] { "ThisIsNotInCompilation.cs", "SomethingElse.cs" });
+        var options = CreateOptions(["ThisIsNotInCompilation.cs", "SomethingElse.cs"]);
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/AnalysisContext/SonarAnalysisContextBaseTest.ShouldAnalyzeTree.cs:69]: Out of curiosity: What made you touch this file?

File: analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.cs [hunk 6]
```diff
@@ -24,1068 +24,1109 @@
...
-        {
-            var code = @"Main(true, 0, out outParameter, ref refParameter);";
-            var context = CreateContextCS(code, additionalParameters: "out int outParameter, ref int refParameter");
-            context.ValidateEntry(LiveIn("refParameter"), LiveOut("refParameter"));
-            context.Validate(code, LiveIn("refParameter"));
-        }
+    [TestMethod]
+    public void ProcessParameterReference_UsedAsOutArgument_NotLiveIn_NotLiveOut()
+    {
+        var code = """Main(true, 0, out outParameter, ref refParameter);""";
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.cs:66]:
```suggestion
        var code = "Main(true, 0, out outParameter, ref refParameter);";
```

File: analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.cs [hunk 7]
```diff
@@ -24,1068 +24,1109 @@
...
-        [TestMethod]
-        public void ProcessParameterReference_InNameOf_NotLiveIn_NotLiveOut()
-        {
-            var code = @"Method(nameof(intParameter));";
-            CreateContextCS(code).ValidateAllEmpty();
-        }
+    [TestMethod]
+    public void ProcessParameterReference_InNameOf_NotLiveIn_NotLiveOut()
+    {
+        var code = """Method(nameof(intParameter));""";
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.cs:75]: Same here

Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.cs:75]: and below

File: analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.cs [hunk 8]
```diff
@@ -24,1068 +24,1109 @@
...
+            Validate(Cfg.EntryBlock, null, expected);
 
-            public void ValidateExit(params Expected[] expected)
-            {
-                expected.All(x => x.Kind == ExpectedKind.Captured).Should().BeTrue("Exit block should expect only Captured variables.");
-                Validate(Cfg.ExitBlock, null, expected);
-            }
+        public void ValidateExit(params Expected[] expected)
+        {
+            Array.TrueForAll(expected, x => x.Kind == ExpectedKind.Captured).Should().BeTrue("Exit block should expect only Captured variables.");
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.cs:1105]: Use `expected.Should().AllSatisfy(...` instead

File: analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.LocalFunction.cs [hunk 9]
```diff
@@ -18,388 +18,408 @@
...
-var usedInFinally = 42;
-var usedInUnreachable = 42;
-if (boolParameter)
-    return;
-LocalFunction();
-
-int LocalFunction()
-{
-    try
+static int LocalFunction(int a) => a + 1;
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.LocalFunction.cs:30]: Alignment is off here

File: analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.TryCatchFinally.cs [hunk 10]
```diff
@@ -21,953 +21,988 @@
...
+        // Finally region
+        context.Validate<IIsNullOperation>("ms = new MemoryStream()", LiveIn("ms"), LiveOut("ms"));     // Null check
+        context.Validate<IInvocationOperation>("ms = new MemoryStream()", LiveIn("ms"));                // Actual Dispose
+        context.Validate(context.Cfg.Blocks[6], null);
+    }
+
+    [TestMethod]
+    public void Using_Nested_Block_LiveInUntilTheEnd()
+    {
+        var code = $$"""
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.TryCatchFinally.cs:68]: I don't see it used
```suggestion
        var code = """
```

File: analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.TryCatchFinally.cs [hunk 11]
```diff
@@ -21,953 +21,988 @@
...
+        context.Validate<IIsNullOperation>("msInner = new MemoryStream()", LiveIn("msInner", "msOuter"), LiveOut("msInner", "msOuter"));   // Null check
+        context.Validate<IInvocationOperation>("msInner = new MemoryStream()", LiveIn("msInner", "msOuter"), LiveOut("msOuter"));          // Actual Dispose
+        context.Validate<IIsNullOperation>("msOuter = new MemoryStream()", LiveIn("msOuter"), LiveOut("msOuter"));                         // Null check
+        context.Validate<IInvocationOperation>("msOuter = new MemoryStream()", LiveIn("msOuter"));                                         // Actual Dispose
+    }
+
+    [TestMethod]
+    public void Using_Nested_Declaration_LiveInUntilTheEnd()
+    {
+        var code = $$"""
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.TryCatchFinally.cs:98]: Same here

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
Polishing

File: analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/LiveVariableAnalysisBase.cs (hunk 3 above)
Comment (mary-georgiou-sonarsource) [reply analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/LiveVariableAnalysisBase.cs:106]: I used "[ ]".
Simple and beautiful.

File: analyzers/src/SonarAnalyzer.CSharp/LiveVariableAnalysis/SonarCSharpLiveVariableAnalysis.cs (hunk 4 above)
Comment (mary-georgiou-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/LiveVariableAnalysis/SonarCSharpLiveVariableAnalysis.cs:107]: Not a big fan of the extra `:` and `case`. Or reads more naturally. But not going to oppose to it.

Comment (mary-georgiou-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/LiveVariableAnalysis/SonarCSharpLiveVariableAnalysis.cs:107]: EDIT: In any case, I think I should not have touched this file.

File: analyzers/tests/SonarAnalyzer.Test/AnalysisContext/SonarAnalysisContextBaseTest.ShouldAnalyzeTree.cs (hunk 5 above)
Comment (mary-georgiou-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/AnalysisContext/SonarAnalysisContextBaseTest.ShouldAnalyzeTree.cs:69]: excessive zeal I think.
I'm reverting.

File: analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.cs (hunk 8 above)
Comment (mary-georgiou-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.cs:1105]: Issue is that expected for some tests is empty, and the `AllSatisfy` is failing if the collection is empty.
Maybe I should divide these tests (with empty and non empty expected) - wdyt?

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**